awesomplete.js

```diff
@@ -262,11 +270,26 @@ _.ITEM = function (text, input) {
 };
 
 _.REPLACE = function (text) {
-	this.input.value = text;
+	this.input.value = text.value;
 };
 
+_.DATA = function (item/*, input*/) { return item; };
+
 // Private functions
 
+function Suggestion(data) {
+	var o = Array.isArray(data)
+	  ? { label: data[0], value: data[1] }
+	  : typeof data === "object" ? data : { label: data, value: data };
+
+	this.label = o.label;
+	this.value = o.value;
+}
+Suggestion.prototype = new String;
```

Comment:
Better to use `Object.create(String.prototype)` here I think, that way the constructor is not called pointlessly. But that’s minor (and more of a general best practice).

Also, I’m not sure if this is in general a good way to subclass builtins, since builtins are special in a number of ways. E.g. I suspect (haven't tested it) that `length` will not work properly so we might need to define our own getter. Check [this](http://www.2ality.com/2013/03/subclassing-builtins-es6.html) out, notably the section about `__proto__`.